packages/app/ios/Orbit/IngestEventEmitter.m

```diff
@@ -0,0 +1,78 @@
+//
+//  EventEmitter.m
+//  Orbit
+//
+//  Created by Ozzie Kirkby on 2022-05-07.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "IngestEventEmitter.h"
+#import <libkern/OSAtomic.h>
+#import <stdatomic.h>
+
+@implementation IngestEventEmitter
+
+@synthesize semaphore;
+
+- (instancetype)init {
+  self = [super init];
+  return self;
+}
+
++ (BOOL)requiresMainQueueSetup {
+    return NO;
+}
+
+-(void)startObserving {
+    self.hasListeners = YES;
+}
+
+// Will be called when this module's last listener is removed, or on dealloc.
+-(void)stopObserving {
+    self.hasListeners = NO;
+}
+
+RCT_EXPORT_MODULE();
+
++ (id)allocWithZone:(NSZone *)zone {
+    static IngestEventEmitter *sharedInstance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [super allocWithZone:zone];
+    });
+    return sharedInstance;
+}
+
+- (NSArray<NSString *> *)supportedEvents {
+    return @[@"onIngestEvent"];
+}
+
+- (bool) emitIngestEvent:(NSString *) fileJSON {
```

Comment:
`- (bool)emitIngestEvent:(NSString *)fileJSON`

Reply:
For [weird historical reasons](https://stackoverflow.com/questions/541289/objective-c-bool-vs-bool), all `bool` should be `BOOL` in Objective-C—they're encoded slightly differently, and in a way Foundation expects.